### PR TITLE
Add contributing guide

### PR DESCRIPTION
**Pull request by Boy132**

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+* [PHP](https://php.net) & [Laravel](https://laravel.com)
+* [Livewire](https://laravel-livewire.com) & [Filament](https://filamentphp.com)
+* [Git](https://git-scm.com) & [Github](https://github.com)
+
+## Dev Environment Setup
+
+1. Fork the Repository
+2. Clone your Fork
+3. Install Dependencies (PHP modules & composer, and run `composer install`)
```
Comment (RMartinOscar) [inline contributing.md:17]:
```suggestion
3. Install Dependencies (PHP modules & composer, and run `composer install --optimize-autoloader`)
```

Comment (RMartinOscar) [inline contributing.md:17]:
```suggestion
3. Install Dependencies (PHP modules & composer, and run `composer install --optimize-autoloader`)
```

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+4. Configure your Environment (via `php artisan p:environment:setup`)
+5. Set up your Database (via `php artisan p:environment:database`) and run Migrations (via `php artisan migrate --seed --force`)
+6. Create your first Admin User (via `php artisan p:user:make`)
+7. Start your Webserver (e.g. Nginx or Apache)
```
Comment (RMartinOscar) [inline contributing.md:21]:
```suggestion
4. Set file permissions
4.1. Allow write (via `chmod 755 -R storage/* bootstrap/cache`)
4.2. Set .gitignores as read only (via `chmod 644 {bootstrap/cache,storage/framework/testing}/.gitignore`)
5. Configure your Environment (via `php artisan p:environment:setup`)
6. Set up your Database (via `php artisan p:environment:database`) and run Migrations (via `php artisan migrate --seed --force`)
7. Create your first Admin User (via `php artisan p:user:make`)
8. Start your Webserver (e.g. Nginx, Apache or Caddy)
```

**Review by RMartinOscar — COMMENTED** [review]
We should specify those to avoid .gitignore files getting mod updates

File: contributing.md (hunk 1 above)
Comment (Boy132) [reply contributing.md:17]: No, in development you should not optimize the autoloader. https://getcomposer.org/doc/articles/autoloader-optimization.md#autoloader-optimization

File: contributing.md (hunk 2 above)
Comment (Boy132) [reply contributing.md:21]: Setting file permissions is only needed on Linux. The guide mainly targets Windows/ macOS users.

File: contributing.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+As IDE we recommend [Visual Studio](https://visualstudio.microsoft.com)/ [Visual Studio Code](https://code.visualstudio.com) (free) or [PhpStorm](https://www.jetbrains.com/phpstorm) (paid).
```
Comment (RMartinOscar) [inline contributing.md:23]:
```suggestion
As IDE we recommend [Visual Studio Code](https://code.visualstudio.com) [Sublime Text](https://sublimetext.com) (free) or [PhpStorm](https://www.jetbrains.com/phpstorm) (paid).
```

I wouldn't recommend using Visual Studio to do anything web related so i'v added Sublime Text as an alternative to VSCode which is also free

Comment (Boy132) [reply contributing.md:23]: Visual Studio works fine for web development. But Sublime text is basically just a lite version of vscode.

**Review by lancepioch — APPROVED** [review]

**Pull request merged**